app/src/main/java/com/github/featuredetectandroid/MainActivity.kt

```diff
@@ -1,27 +1,138 @@
+import androidx.camera.core.ImageAnalysis
+import androidx.camera.core.ImageAnalysis.OUTPUT_IMAGE_FORMAT_YUV_420_888
+import androidx.camera.lifecycle.ProcessCameraProvider
 import androidx.compose.foundation.layout.Box
 import androidx.compose.foundation.layout.fillMaxSize
-import androidx.compose.material.FloatingActionButton
-import androidx.compose.ui.Alignment
+import androidx.compose.material.Text
 import androidx.compose.ui.Modifier
+import androidx.core.content.ContextCompat
+import com.github.featuredetectandroid.ui.GrayscaleView
+import com.github.featuredetectandroid.ui.GrayscaleViewModel
 import com.github.featuredetectandroid.ui.theme.Theme
+import com.github.featuredetectandroid.utils.PhotoAnalyzer
+import java.util.concurrent.Executors
+
+private const val TAG = "MainActivity"
 
 class MainActivity : ComponentActivity() {
+    private val cameraExecutor = Executors.newSingleThreadExecutor()
+    private val imageViewModel by viewModels<GrayscaleViewModel>()
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+
+        tryStartCamera()
         setContent {
             Theme {
                 Box(Modifier.fillMaxSize()) {
-                    FloatingActionButton(
-                        onClick = { /*TODO*/ },
-                        modifier = Modifier.align(Alignment.BottomEnd)
-                    ) { }
+                    if (ContextCompat.checkSelfPermission(
+                            this@MainActivity,
+                            Manifest.permission.CAMERA
+                        ) == PackageManager.PERMISSION_GRANTED
+                    ) {
+                        GrayscaleView(
+                            imageViewModel.grayscaleBitmap,
+                            imageViewModel.width,
+                            imageViewModel.height
+                        )
+                    } else {
+                        Text(text = "Camera permission required")
+                    }
                 }
             }
         }
     }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        cameraExecutor.shutdown()
+    }
+
+    private val cameraPermission =
+        registerForActivityResult(ActivityResultContracts.RequestPermission()) { granted ->
+            if (granted) {
+                Toast.makeText(
+                    this,
+                    "Permissions granted.",
+                    Toast.LENGTH_SHORT
+                ).show()
+                Log.d(TAG, "Permission was granted successfully.")
+                startCamera()
+            } else {
+                Toast.makeText(
+                    this,
+                    "Permissions not granted by the user.",
+                    Toast.LENGTH_SHORT
+                ).show()
+            }
+        }
+
+    private fun tryStartCamera() {
+        when {
+            cameraPermissionGranted() -> {
+                startCamera()
+            }
```

Comment:
Braces are not needed here

Reply:
> Braces are not needed here

Wouldn't it be better to use braces here for reasons of uniformity?

Reply:
I think, it's more readable without additional braces, because it takes less space. Style guide also seems to [permit this](https://developer.android.com/kotlin/style-guide#braces).

Reply:
fixed